### PR TITLE
Simple EBNF changes and Added Variable LUT

### PR DESCRIPTION
**Pull request by A-Russell23**
# Simple EBNF
## Changes:
- Changed `Expression`  so you can have more than one extra + or minus followed by `Term`
- Changed `Term` so you can more than one multiply or divide followed by `Factor`
- Ensured an `Identifier` can be assigned to another `Identifier`
- Ensured `Identifier` can be used in an expression by adding it to the definition of `Factor`

# Added Variable LUT
## Changes:
- Added `VariableTable.cs`
- In `Parser.cs` changed `Statement()` to check if the identifier exists and is of the correct type before assigning a variable
- In `Parse.cs` changed `DeclareVariable()` to check the identifier doesn't already exist and then add it to the LUT

### CODE REVIEW

**Review by KThompson-Lane — DISMISSED** [review]
All looks good to me 👍

**Review by Spartan-Bailey — APPROVED** [review]

File: C_Flat_Interpreter/Common/VariableTable.cs [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+public class VariableTable
+{
+    private Dictionary<string, ParseNode> _table;
+    public VariableTable()
+    {
+        _table = new Dictionary<string, ParseNode>();
+    }
+
+    public void Add(string word, ParseNode node)
```
Comment (KThompson-Lane) [inline C_Flat_Interpreter/Common/VariableTable.cs:14]: These two `Add` methods could potentially be merged into one, with an optional parameter for node. e.g.:
`public void Add(string word, ParseNode node = new ParseNode(NodeType.Null)` 
or something along those lines, it does make the method signature a little more complicated though so happy for you to decide on this one :)

**Review by KThompson-Lane — APPROVED** [review]
Had a few minor notes but looks good to me overall :D

File: C_Flat_Interpreter/Common/VariableTable.cs (hunk 1 above)
Comment (A-Russell23) [reply C_Flat_Interpreter/Common/VariableTable.cs:14]: Thats something I can look into as part of my clean up work

**Pull request merged**